app/controllers/responses_controller.rb

```diff
@@ -22,6 +21,13 @@ def update
 
   private
 
+  def response_by_show_secret_or_id
+    return Response.find_by_show_secret params[:show_secret] if params[:show_secret]
+    return Response.find params[:id] if params[:id]
+
```

Comment:
Maakt dit uit? De arts weet toch niet wat de `show_secret` is totdat hij een e-mail krijgt om de response te bekijken.

En alleen de invuller kan de `id` weten, en het is ook de bedoeling dat hij de `show`-actie kan gebruiken om een response op te vragen. Dat is namelijk nodig om de vervolgvraag op te halen tijdens het invullen

Als je deze opbouw onhandig vindt of alsnog een security-probleem ziet hoor ik het graag, maar voorlopig laat ik het zo.